src/Mutex/Impl/ILockHandle.cs

```diff
@@ -0,0 +1,10 @@
+namespace TemporalioSamples.Mutex.Impl;
```

Comment:
I think `ILockHandle` and `WorkflowMutex` can move up to the `TemporalioSamples.Mutex` with the rest (can make former inner class of latter if wanting fewer files). And can probably move `MutexWorkflow.workflow.cs` up there too and put most things it uses as inner classes if we want to avoid file sprawl.

Your call, but I think there's value in fewer files/namespaces here

Reply:
I wanted everything related to the implementation of the mutex to be clearly separated, meaning that potentially it could be a separate project, where only WorkflowMutex.LockAsync (and possibly methods that register workflows/activities) is accessible. On the other hand, since this is just an example, creating multiple projects is overkill

Reply:
Hrmm, yeah, makes sense to separate everything out, just maybe not two levels deep then. I wish I could think of a better word than `Impl` heh.